homeassistant/components/rfxtrx.py

```diff
@@ -31,6 +38,42 @@
 _LOGGER = logging.getLogger(__name__)
 RFXOBJECT = None
 
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_PACKETID): cv.slug,
+    vol.Optional(ATTR_FIREEVENT, default=False): cv.boolean,
+})
+
+
+def _valid_device(value):
+    """Validate a dictionary of devices definitions."""
+    config = OrderedDict()
+    for key, device in value.items():
+        try:
+            key = cv.slug(key)
+            config[key] = DEVICE_SCHEMA(device)
+        except vol.MultipleInvalid as ex:
+            raise vol.Invalid('Rfxtrx device {} is invalid: {}'
+                              .format(key, ex))
+    return config
+
+
+DEFAULT_SCHEMA = vol.Schema({
+    vol.Required("platform"): DOMAIN,
+    vol.Required(CONF_DEVICES): vol.All(dict, _valid_device),
```

Comment:
btw, slug is used for values that can be put in urls / entity ids. It will even suggest to the user what they should use instead. I think for packet ids it would be best to write your own validator that validates it's numbers/letters and the correct length (or range)